app/src/androidTest/java/doit/study/droid/data/local/dao/DbTest.kt

```diff
@@ -0,0 +1,45 @@
+package doit.study.droid.data.local.dao
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import doit.study.droid.data.local.QuizDatabase
+import doit.study.droid.data.local.dao.util.MainCoroutineRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+
+abstract class DbTest {
+
+    private lateinit var database: QuizDatabase
+
+    val db: QuizDatabase
+        get() = database
+
+    // Set the main coroutines dispatcher for unit testing.
+    @ExperimentalCoroutinesApi
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    // Executes each task synchronously using Architecture Components.
+    @get:Rule
+    var instantExecutorRule = InstantTaskExecutorRule()
+
+    // That rule conflicts with other rules and throws RuntimeException: Delegate runner
+    // @Rule
+    // var thrown: ExpectedException = ExpectedException.none()
```

Comment:
I think, all comments in this test isn't necessary - code is self documented already.

Reply:
@rinekri Here is a conflict of different rules (sort of a reminder why old technics is used to handle exceptions)